crates/core/src/transport/peer_connection.rs

```diff
@@ -365,23 +366,93 @@ async fn packet_sending(
     payload: impl Into<SymmetricMessagePayload>,
     sent_tracker: &parking_lot::Mutex<SentPacketTracker<InstantTimeSrc>>,
 ) -> Result<()> {
-    // FIXME: here ensure that `confirm_receipt` won't make the packet exceed the max data size
-    // if it does, split it to send multiple noop packets with the receipts
-
-    // tracing::trace!(packet_id, "sending packet");
-    let packet = SymmetricMessage::serialize_msg_to_packet_data(
+    match SymmetricMessage::try_serialize_msg_to_packet_data(
         packet_id,
         payload,
         outbound_sym_key,
         confirm_receipt,
-    )?;
-    outbound_packets
-        .send((remote_addr, packet.clone().prepared_send()))
-        .await
-        .map_err(|_| TransportError::ConnectionClosed)?;
-    sent_tracker
-        .lock()
-        .report_sent_packet(packet_id, packet.prepared_send());
+    )? {
+        either::Either::Left(packet) => {
+            outbound_packets
+                .send((remote_addr, packet.clone().prepared_send()))
+                .await
+                .map_err(|_| TransportError::ConnectionClosed)?;
+            sent_tracker
+                .lock()
+                .report_sent_packet(packet_id, packet.prepared_send());
+            Ok(())
+        }
+        either::Either::Right((payload, confirm_receipt)) => {
+            // FIXME: how to split confirm_receipt into multiple msgs efficiently?
+            let mut packets = Vec::with_capacity(8);
```

Comment:
the optimal packet size is to get a noop packet which contains as many receipts which fit into MAX_DATA_SIZE - overhead of the SymmetricMessage serialisation